### ISSUE TITLE
CA1062 ignores null-forgiving operator used on argument of type Type

### ISSUE THREAD

**Issue by lonix1** [dotnet/roslyn-analyzers#5350]
### Analyzer

**Diagnostic ID**: [CA1062](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1062): `Validate arguments of public methods`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.301](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

````cs
public void DoSomething(Type type) {
  var method = type!.GetMethod("Foo");            // <--- analyzer triggered here
  if (method == null) throw new Exception();
  // ...or carry on
}
````

### Steps To Reproduce

See above.

### Expected behavior

I use the null forgiving operator `type!` so CA1062 should not be triggered.

### Actual behavior

CA1062 is triggered.

## Additional context

no

**Comment by Youssef1313** [comment]
Why do you expect `type` to be non-null? It's a public method so it can indeed be null.

**Comment by lonix1** [comment]
If I use that exact code with a different type, and I use the null-forgiving operator `!` then the analyzer doesn't complain.

But with type `Type` is does complain.

**Comment by mavasani** [comment]
@Youssef1313 is correct, this is by design.
